### ISSUE TITLE
[bitnami/mongodb] ReplicaSet scale down issue

### ISSUE THREAD

**Issue by emikolajczak** [bitnami/charts#3073]
**Which chart**:
MongoDB 8.0.9

**Describe the bug**
There is an issue when scale down of MongoDB ReplicaSet. ReplicaSet is not re-configured, deleted mongo nodes are not being
removed from ReplicaSet configuration. `rs.status()` return it as `"(not reachable/healthy)"`. In my opinion deleted nodes should be removed from ReplicaSet, because can cause lack of quorum.
Everything is ok when scale up, new mongo node is being added to ReplicaSet config.

**To Reproduce**
1. Deploy 1 PRIMARY + 2 SECONDARY + ARBITER ReplicaSet
2. Scale down to 1 PRIMARY + 1 SECONDARY + ARBITER ReplicaSet
3. ReplicaSet configuration not changed.

**Expected behavior**
Deleted mongo nodes removed from ReplicaSet

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.1.0", GitCommit:"b29d20baf09943e134c2fa5e1e1cab3bf93315fa", GitTreeState:"clean", GoVersion:"go1.13.7"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.5", GitCommit:"e0fccafd69541e3750d460ba0f9743b90336f24f", GitTreeState:"clean", BuildDate:"2020-04-16T11:44:03Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.5", GitCommit:"e0fccafd69541e3750d460ba0f9743b90336f24f", GitTreeState:"clean", BuildDate:"2020-04-16T11:35:47Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by andresbono** [comment]
Hi @emikolajczak, how are you scaling the cluster? You will need to execute `rs.remove("<the_node>")` [doc](https://docs.mongodb.com/manual/tutorial/remove-replica-set-member/#remove-a-member-using-rs-remove). This is what we tried:

`custom_values.yaml`
```
architecture: replicaset
replicaCount: 3
auth:
  rootPassword: mypass
  replicaSetKey: myrskey
```

Install the chart
```
$ helm install mongo bitnami/mongodb -f custom_values.yaml
```

Scale down the replicas and delete missing member:
```
$ helm upgrade mongo bitnami/mongodb -f custom_values.yaml --set replicaCount=2
$ kubectl run --namespace default mongo-mongodb-client --rm --tty -i --restart='Never' --image docker.io/bitnami/mongodb:4.2.8-debian-10-r26 --command -- mongo admin --host "mongo-mongodb-0.mongo-mongodb-headless.default.svc.cluster.local" --authenticationDatabase admin -u root -p mypass --eval 'rs.remove("mongo-mongodb-2.mongo-mongodb-headless.default.svc.cluster.local:27017")'
```

**Comment by pawcykca** [comment]
Hi @andresbono 

All this steps works ok for scaling down, but after when we are trying to scale up replica again the member that was removed from replica is not added automatically and we need to add it manually executing `rs.add("...")`

**Comment by emikolajczak** [comment]
> Hi @andresbono
> 
> All this steps works ok for scaling down, but after when we are trying to scale up replica again the member that was removed from replica is not added automatically and we need to add it manually executing `rs.add("...")`

Indeed. Same issue here. After scale down member is not automatically added to replica during next scale up.

**Comment by andresbono** [comment]
Correct, if you explicitly remove a member from the cluster to avoid the `"(not reachable/healthy)"` state, you will need to add them again manually when scaling up.

We are going to check if there is something we can do to improve the user experience. Please, any suggestions and contributions from you to improve this chart are welcome!

**Comment by juan131** [comment]
Hi @pawcykca @emikolajczak 

The problem here is that we cannot distinguish (at container/pod level) when a MongoDB container is shutting down due to an upgrade, scaling down the solution or a K8s node being drained. So, we cannot use "pre-stop hooks" to remove the node from the cluster.  Something very similar happened on RabbitMQ, and we had to document in the **README.md** how to "forget" members in the cluster when scaling down:  

>When scaling down the solution unnecessary RabbitMQ nodes are automatically stopped, but they are not removed from the cluster. You need to manually remove them running the `rabbitmqctl forget_cluster_node` command. For instance, if you initially installed RabbitMQ with 3 replicas and then you scaled it down to 2 replicas, run the commands below (assuming that the release name is `rabbitmq` and you're using `hostname` as clustering type):
>
>```console
>$ kubectl exec rabbitmq-0 --container rabbitmq -- rabbitmqctl forget_cluster_node rabbit@rabbitmq-2.rabbitmq->headless.default.svc.cluster.local
>$ kubectl delete pvc data-rabbitmq-2
>```

So basically, what's happening to you is that you're removing the member from the replicaset, but you're not removing the associated pod's PVC. Therefore, when you scale up the solution again, it reuses the previous PVC which has data that assumes the member was never removed from the replicaset and, therefore, the conflict. 

Please, ensure that you remove the PVC, after running the command suggested by @andresbono to remove the replicaset member.

**Comment by emikolajczak** [comment]
Hi @juan131 
Your solution is fine. After remove PVC and add member again replicaSet is correctly configured.

**Comment by andresbono** [comment]
Thanks for confirming @emikolajczak.

**Comment by sniperking1234** [comment]
Is there an automated solution for this operation now?
